### DISCUSSION TITLE
Adding new files

### DISCUSSION THREAD

**Discussion by peterbartels** [codesandbox/sandpack#503] [Q&A] [answered]
What is the best way to add new files?
There are methods such as deleteFile and updateFile but I don't see a method for adding files.

Adding files to the files={...} prop seems to work but I'm not sure if it is the corect way of doing it.

**Comment by danilowoz** [comment, marked as answer]
Hey, what is precisely the feature desired?
A) User insertion: the user be able to create new files and possibly folders to insert a new piece of code?
B) Extend the template files: add new files beyond the template files; for example, add a button.tsx?  

So, _A_ is still not possible, and we haven't had the chance to explore this possibility, but it's certainly doable. But if the case is _B_, you can do it using the `files` prop, as you mentioned.

Please, let me know what's exactly you're looking for, and then we can iterate it into a new feature 😄

**Reply by peterbartels** [reply]
I'm building a component library creator/manager and I want the user to be able to create new components and therefor can create files and directories.
Because `SandpackFileExplorer ` doesn't have all the functionality I need, I have implemented a custom FileTree component that uses the `useSandpack` hook to manage the state of sandpack. The hook doesn't have a method for adding files.

**Reply by wengjq** [reply]
I have the same problem

**Comment by wengjq** [comment]
I think the SandbackFileExplorer component should have the function of adding, deleting and modifying files.like the website https://codesandbox.io/
